File: AOJ/0280/0280.cpp
#include<iostream>
#include<string>
#include<cstdlib>
#include<algorithm>
using namespace std;
#define MAX 10

void play(string line, int N){
  int C[MAX];
  int B = 0;
  for (int i = 0; i < N; i++ ) C[i] = 0;

  for ( int i = 0; i < line.size(); i++ ){
    int p = i%N;
    if ( line[i] == 'M' ){
      C[p]++;
    } else if ( line[i] == 'S' ){
      C[p]++;
      B += C[p];
      C[p] = 0;
    } else if ( line[i] == 'L' ){
      C[p] += B + 1;
      B = 0;
    }
  }

  sort(C, C+N);

  for ( int i = 0; i < N; i++ ) cout << C[i] << " ";
  cout << B << endl;
}

main(){
  int N;
  string line;
  while(1){
    cin >> N;
    if ( N == 0 ) break;
    cin >> line;
    play(line, N);
  }
}

File: AOJ/0280/0280.py
def play(line, N):
    C = [0] * N
    B = 0

    for i in range(len(line)):
        p = i%N
        if line[i] == 'M':
            C[p] += 1
        elif line[i] == 'S':
            C[p] += 1
            B += C[p]
            C[p] = 0
        elif line[i] == 'L':
            C[p] += B + 1
            B = 0

    C.sort();

    for c in C: print(c, end=" ")
    print(B)

while True:
    N = int(input())
    if N == 0: break
    line = input()
    play(line, N)

